### PR TITLE
Use `repository.directory` field in `package.json` files

### LINKED ISSUE #11619: Modify all our package.json's to use a different value for the repository field
So `package.json` files can have a `repository` field!

After looking at https://github.com/babel/babel-polyfills with @nicolo-ribaudo, we realized there's another format for it that we should change all the packages in this monorepo to. https://docs.npmjs.com/files/package.json#repository

image

So for example: `babel-cli`

```json
"repository": "https://github.com/babel/babel/tree/master/packages/babel-cli",
```
we should change to:

```json
"repository": {
  "type" : "git",
  "url" : "https://github.com/babel/babel.git",
  "directory": "packages/babel-cli"
}
```

I would change all the files in eslint/codemods/packages folders!

### PR DESCRIPTION
**Pull request by saulosantiago**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #11619  
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

Changed all packages.json's files, adding the new repository field for sub-packages.

### CODE REVIEW

File: eslint/babel-eslint-config-internal/package.json [hunk 1]
```diff
@@ -6,7 +6,11 @@
   "homepage": "https://babeljs.io/",
   "license": "MIT",
   "private": true,
-  "repository": "https://github.com/babel/babel/tree/master/eslint/babel-eslint-config-internal",
+  "repository": {
+    "type" : "git",
+    "url" : "https://github.com/babel/babel.git",
+    "directory": "packages/babel-eslint-config-internal"
```
Comment (nicolo-ribaudo) [inline eslint/babel-eslint-config-internal/package.json:12]:
```suggestion
    "directory": "eslint/babel-eslint-config-internal"
```

File: packages/babel-plugin-transform-react-pure-annotations/package.json [hunk 2]
```diff
@@ -2,7 +2,11 @@
   "name": "@babel/plugin-transform-react-pure-annotations",
   "version": "7.10.0",
   "description": "Mark top-level React method calls as pure for tree shaking",
-  "repository": "https://github.com/babel/babel/tree/master/packages/babel-plugin-transform-react-pure",
+  "repository": {
+    "type" : "git",
+    "url" : "https://github.com/babel/babel.git",
+    "directory": "packages/babel-plugin-transform-react-pure"
```
Comment (nstepien) [inline packages/babel-plugin-transform-react-pure-annotations/package.json:8]:
```suggestion
    "directory": "packages/babel-plugin-transform-react-pure-annotations"
```

**Comment by nicolo-ribaudo** [comment]
Thanks!

**Comment by saulosantiago** [comment]
Thanks for give me this opportunity to contribute :)

**Pull request merged**